CallMeL/Empathetic-Dialogue-System
# Understand nano GPT

## Motivation and Context

Understand nanoGPT so we can better train the model
## What's new in this PR
1. Add [ detailed introduction readme](https://github.com/CallMeL/Project-ML/blob/532d17a32c01f8df4c28824bde1ec115fda45e82/src/models/nanoGPT/README.md) for nanoGPT
2. In `train.py`, save training process so we can visualize it
3. In `train.py`, doing some modification so now it can load our emotion data
4. Add `chat.py` so we can chat with the model more than randomly generate  new text


## How to test your changes

Train the model and check the output
```
cd ./src/models/nanoGPT

time python train.py \
  --n_layer=4 \
  --n_head=4 \
  --n_embd=64 \
  --compile=False \
  --eval_iters=1 \
  --block_size=64 \
  --batch_size=8 \
  --device=mps
```

(add `--init_from=resume \` for continue training)
